src/models/room.js

```diff
@@ -1312,6 +1302,31 @@ Room.prototype.addPendingEvent = function(event, txnId) {
 
     this.emit("Room.localEchoUpdated", event, this, null, null);
 };
+
+Room.prototype._savePendingEvents = function() {
+    const pendingEvents = this._pendingEventList.map(event => {
+        return {
+            ...event.event,
+            txn_id: event.getTxnId(),
+        };
+    }).filter(event => {
+        // Filter out the unencrypted messages if the room is encrypted
```

Comment:
Hmm, so why is this filtering what we want? I think I am just a little unsure what the expected behaviour for this persistence is in an encrypted room... I guess we could have both unencrypted and encrypted events queued up for an encrypted room, depending on whether we encrypted them before persisting...? Maybe we need a few comments on what's expected...

Reply:
Fair enough, 
So at this point this is my understanding of how things work
- In an unencrypted room, all messages are unencrypted
- In an encrypted room, all messages are encrypted

The flow in that `Room` model is a bit odd as a message, will have the following flow, right after a user hits send:
1. `EVENT_STATUS=sending` (message unencrypted)
2. `EVENT_STATUS=encrypting` (message unencrypted)
3. `EVENT_STATUS=sending` (message encrypted)
4. `EVENT_STATUS=sent`

on an unencrypted room, the steps 2 and 3 are skipped.
It is a bit difficult in that model to know at which stage we are as `EVENT_STATUS` in step 1 and 3 are the same.
It seems like a sensible thing to do to not persist events in `localStorage` if they are unencrypted in a room that requires encryption (the use case that @t3chguy raised)

So I can definitely add comments in code to highlight my reasoning, but please do let me know if I'm missing a key part of the logic

Reply:
Hmm okay, right, that helps me follow the context a bit better. Seems good indeed to do this kind of filtering.